src/effects/VST3/VST3Effect.h

```diff
@@ -0,0 +1,153 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3Effect.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+
+#pragma once
+
+#include <pluginterfaces/gui/iplugview.h>
+#include <pluginterfaces/vst/ivstaudioprocessor.h>
+#include <public.sdk/source/vst/hosting/module.h>
+
+#include "EffectInterface.h"
+
+#include "SampleCount.h"
+#include "effects/Effect.h"
+#include "internal/ComponentHandler.h"
+
+class NumericTextCtrl;
+
+namespace Steinberg
+{
+   namespace Vst
+   {
+      class IComponent;
+      class IEditController;
+      class IParameterChanges;
+      class IComponentHandler;
+      class IConnectionPoint;
+   }
+}
+
+class ParameterChangesProvider;
+
+/**
+ * \brief Objects of this class connect Audacity with VST3 effects
+ */
+class VST3Effect final : public EffectUIClientInterface
+{
+   //Keep strong reference to a module while effect is alive
+   std::shared_ptr<VST3::Hosting::Module> mModule;
+
+   //Following fields are unique to each effect instance
+
+   Steinberg::IPtr<Steinberg::Vst::IComponent> mEffectComponent;
+   Steinberg::IPtr<Steinberg::Vst::IAudioProcessor> mAudioProcessor;
+   Steinberg::IPtr<Steinberg::Vst::IConnectionPoint> mComponentConnectionProxy;
+   Steinberg::IPtr<Steinberg::Vst::IConnectionPoint> mControllerConnectionProxy;
```

Comment:
I think the two fields above are non-null only when the object is not constructed by copy and should be in the section with `mPlugView` etc., below.